### ISSUE TITLE
upsert does not ignore comparison operators in the query 

### ISSUE THREAD

**Issue by garhiggins** [mongomock/mongomock#451]
From the MongoDB Docs (https://docs.mongodb.com/manual/reference/method/db.collection.update/#upsert-behavior):
```
The update creates a base document from the equality clauses in the <query> parameter, and then applies the update expressions from the <update> parameter. **Comparison operations** from the <query> **will not be included** in the new document.
```
Example:
```
query = {'$or': [{'name': 'billy'}, {'name': 'Billy'}]}
update = {'$set': {'name': 'Billy', 'age': 5}}
db.widgets.update(query, update, upsert=True)
```
This should insert `{name: 'Billy', age: 5}`, but instead results in the following exception:
```
  File "/Users/garhiggins/cambly/Cambly-Backend/venv/lib/python2.7/site-packages/mongomock/collection.py", line 535, in update
    check_keys, **kwargs)
  File "/Users/garhiggins/cambly/Cambly-Backend/venv/lib/python2.7/site-packages/mongomock/collection.py", line 822, in _update
    upserted_id = self._insert(existing_document)
  File "/Users/garhiggins/cambly/Cambly-Backend/venv/lib/python2.7/site-packages/mongomock/collection.py", line 463, in _insert
    BSON.encode(data, check_keys=True)
  File "/Users/garhiggins/cambly/Cambly-Backend/venv/lib/python2.7/site-packages/bson/__init__.py", line 1027, in encode
    return cls(_dict_to_bson(document, check_keys, codec_options))
bson.errors.InvalidDocument: key '$or' must not start with '$'
```
If I print `data` from L463 of collection.py, I find that the comparison operation from the query has been included in the document:
`{'$or': [{'name': 'billy'}, {'name': 'Billy'}], 'age': 5, '_id': ObjectId('5ba99a4f5c1f751e7e962eb7'), 'name': 'Billy'}`

**Comment by garhiggins** [comment]
FWIW, this was a regression in 3.9.0 (we're currently on 3.8.0, and i can't update due to this).

**Comment by pcorpet** [comment]
Thanks for reporting @garhiggins and sorry for the regression. I'll have a look.

**Comment by pcorpet** [comment]
This is due to 126b95e, introduced to fix #416. I think that before that the upserted document would actually store the `'$or': [{'name': 'billy'}, {'name': 'Billy'}]` silently.

@garhiggins you should be able to update to v3.10 already while I fix this.

**Comment by pcorpet** [comment]
Ha, got bitten by the comparator `{'$eq': 'billy'}` which is not considered as a comparison operator and as such its value should still be upserted.
